src/main/resources/hazelcast-config-schema.json

```diff
@@ -1373,20 +1373,535 @@
+              "type": "boolean",
+              "default": false
+            },
+            "class-name": {
+              "type": "string"
+            },
+            "properties": {
+              "type": "object"
+            }
+          }
+        },
+        "symmetric-encryption": {
+          "type": "object",
+          "additionalProperties": false,
+          "description": "Lets you encrypt the entire socket level communication among all Hazelcast members. This feature is available only in Hazelcast Enterprise.  Its configuration contains the encryption properties and the same configuration must be placed to all members.",
+          "properties": {
+            "enabled": {
+              "type": "boolean",
+              "default": false
+            },
+            "algorithm": {
+              "type": "string",
+              "default": "PBEWithMD5AndDES"
+            },
+            "salt": {
+              "type": "string",
+              "default": "thesalt"
+            },
+            "password": {
+              "type": "string",
+              "default": "thepassword"
+            },
+            "iteration-count": {
+              "type": "integer",
+              "default": 19
+            }
+          }
+        },
+        "socket-options": {
+          "type": "object",
+          "additionalProperties": false,
+          "properties": {
+            "buffer-direct": {
+              "type": "boolean",
+              "default": false,
+              "description": "Specifies whether direct or non-direct buffers should be allocated for the socket."
+            },
+            "tcp-no-delay": {
+              "type": "boolean",
+              "default": true,
+              "description": "Specifies whether Nagle's algorithm should be disabled. The default value is true meaning that Nagle's algorithm is disabled."
+            },
+            "keep-alive": {
+              "type": "boolean",
+              "default": true,
+              "description": "Specifies whether TCP keep-alive should be enabled."
+            },
+            "connect-timeout-seconds": {
+              "type": "integer",
+              "default": 0,
+              "description": "Specifies the TCP timeout in seconds."
+            },
+            "send-buffer-size-kb": {
+              "type": "integer",
+              "default": 128,
+              "description": "Specifies the size of the send buffer."
+            },
+            "receive-buffer-size-kb": {
+              "type": "integer",
+              "default": 128,
+              "description": "Specifies the size of the receive buffer."
+            },
+            "linger-seconds": {
+              "type": "integer",
+              "default": 0,
+              "description": "Specifies the TCP linger seconds."
+            }
+          }
+        }
+      }
+    },
+    "SocketEndpointConfig": {
+      "$ref": "#/definitions/WanEndpointConfig",
+      "properties": {
```

Comment:
Reading the example yaml file, it seems to have more properties: outbound-ports, interfaces,  ssl, socket-interceptor, symmetric-encryption, socket-options (but maybe I'm just looking at the wrong place)